src/script/lua_api/l_object.cpp

```diff
@@ -1249,7 +1239,7 @@ int ObjectRef::l_set_attribute(lua_State *L)
 		return 0;
 
 	std::string attr = luaL_checkstring(L, 2);
-	if (lua_isnil(L, 3)) {
+	if (lua_isnoneornil(L, 3)) {
```

Comment:
1) This is a deprecated function. Legacy code should not be touched in general.
2) `set_attribute("test")` is less obvious than `set_attribute("test", nil)`, and avoids accidental erasing of metadata when the argument is forgotten by the modder.